docs/fieldnorm/verification_validation.md

```diff
@@ -1,7 +1,86 @@
+> Verification and Validation
+> The verification and validation (V&V) activities should be targeted towards the criticality and impact to patient safety of the SaMD, as discussed in IMDRF/SaMD WG/N12.Typically, verification (providing assurance that the design and development activity at each development stage conforms to the requirements) and validation (providing reasonable confidence that the software meets its intended use/user needs and operational requirements) activities ensure that all elements from the SaMD design and development—including any changes made during maintenance/upgrades—have been implemented correctly and that objective evidence of this implementation is recorded.
+
+In short, Barry Boehm expressed the difference between the software verification and software validation as follows:
+
+> Verification: Are we building the product right?
+> Validation: Are we building the right product?
+
+Hence, for each release _Software verification_ **and** _Software validation_ is required. 
+
+Often these definitions do not match with the definitions used in the medical laboratory. However, in terms of software development and thus bioinformatics, following software verification and software validation definitions make more sense. Noteworthy, the ISO-15189 states that validation is needed for laboratory designed or developed methods and for validated methods subsequently modified. Both are true for software development.
+
+Verification and Validation of software should be performed as described below. 
+
+The IEC-62304 mentiones different aspects of verification; i.e. software unit verification and software integration testing, and validation; i.e. software system test.
+
+
+
+
+## Software verification
+
+Testing whether the code is correctly developed can be done on different levels, for example through:
+
+- Code reviews 
+- Unit testing
+- Integration testing
+- End to end workflow tests
+
+The extend of software verification performed by the Bioinformaticians will differ in each team, it should be clear from the procedure which software verification should be performed. 
+
+## Software validation
+
+In order to support the repeatability of the validation, the following must be registered:
+
+- The person executing the test
+- The person evaluating the test
+- Date of the test
+- The software tested, including its version number
+- If applicable: configuration, hardware, testing tools
+- The test, including steps to take the test, expected results and acceptance criteria
+- The results, including samples used
+- The evaluation of the test
+- Conclusion of the test
+
+The person executing the test should be independent from the person who developed the software and wrote the validation. Software validation is preferably executed by an end user. For example, someone from the laboratory when the end result is mostly used by the laboratory, or a bioinformatician when bioinformatics expertise is needed for performing the validation. 
```

Comment:
Why should the person executing be different from the developer and why is the developer and the person that writes the validation the same person? In our team it is often the case the executor is the developer and he/she also writes the validation (authorized by somebody else though)

Reply:
I looked at the 62304; "Readers of the standard are encouraged to use
IEC 61508 as a source for good software methods, techniques and tools, while recognising that
other approaches, both present and future, can provide equally good results. This standard
makes no recommendation concerning independence of people responsible for one software
ACTIVITY (for example VERIFICATION) from those responsible for another (for example design). In particular, this standard makes **no requirement for an independent safety assessor, since
this is a matter for ISO 14971.**"  <- not sure if "this standard" now refers to the 61508 or the 62304

ISO 14971: "The risk management plan should identify the personnel with responsibility for the execution of specific risk management activities, for example reviewer(s), expert(s), **independent verification specialist(s)**, individual(s) with approval authority (see 3.2). This assignment can be included in a resource allocation matrix defined for the design project. "

Reply:
from a quality point of view, having an independent end user performing the tests would be best. However, in cases where the developer will also function as the end user, which is quite often the case in bioinformatics, it wouldn't make sense for the person to also be independent. 

Maybe we should make independent preferable and end-user a must?

Reply:
My end users do not know how to perform unit tests. It is much more workable to have the developer develop AND run the tests. Then a colleague can review if the code is any good and if the testing was properly reported (and thus executed).

Reply:
unit testing and code review is part of the verification step. This is mainly talking about the validation of the software.

basically, during software verification you look at the technical side of things while for a validation you look at the practical side.

Reply:
This has been discussed in the meeting of 9-2-2024. We have decided that (for now) the validation **should** be performed by an end user and **preferably** by an independent person.